File: samples/client/3_0_3_unit_test/java/src/test/java/org/openapijsonschematools/client/components/schemas/BySmallNumberTest.java
package org.openapijsonschematools.client.components.schemas;

import org.junit.Test;
import org.openapijsonschematools.client.configurations.JsonSchemaKeywordFlags;
import org.openapijsonschematools.client.configurations.SchemaConfiguration;
import org.openapijsonschematools.client.exceptions.ValidationException;
import org.openapijsonschematools.client.exceptions.InvalidTypeException;

public class BySmallNumberTest {
    static final SchemaConfiguration configuration = new SchemaConfiguration(JsonSchemaKeywordFlags.ofNone());

    @Test
    public void test000751IsNotMultipleOf00001Fails() {
        // 0.00751 is not multiple of 0.0001
        final var schema = BySmallNumber.BySmallNumber1.getInstance();
        try {
            schema.validate(
                0.00751d,
                configuration
            );
            throw new RuntimeException("A different exception must be thrown");
        } catch (ValidationException | InvalidTypeException ignored) {
            ;
        }
    }

    @Test
    public void test00075IsMultipleOf00001Passes() {
        // 0.0075 is multiple of 0.0001
        final var schema = BySmallNumber.BySmallNumber1.getInstance();
        schema.validate(
            0.0075d,
            configuration
        );
    }
}

File: samples/client/3_0_3_unit_test/kotlin/src/test/kotlin/org/openapijsonschematools/client/components/schemas/BySmallNumberTest.kt
package org.openapijsonschematools.client.components.schemas

import org.openapijsonschematools.client.configurations.JsonSchemaKeywordFlags
import org.openapijsonschematools.client.configurations.SchemaConfiguration
import org.openapijsonschematools.client.exceptions.ValidationException

import kotlin.test.Test
import kotlin.test.assertFailsWith

class BySmallNumberTest {
    companion object {
        val configuration = SchemaConfiguration(JsonSchemaKeywordFlags.Builder().format().build())
    }

    @Test
    fun test000751IsNotMultipleOf00001Fails() {
        // 0.00751 is not multiple of 0.0001
        val schema = BySmallNumber.BySmallNumber1.getInstance()
        assertFailsWith<ValidationException>(
            block = {
                schema.validate(
                    0.00751,
                    configuration
                )
            }
        )
    }

    @Test
    fun test00075IsMultipleOf00001Passes() {
        // 0.0075 is multiple of 0.0001
        val schema = BySmallNumber.BySmallNumber1.getInstance()
        schema.validate(
            0.0075,
            configuration
        )
    }
}
